File: python/components/gilded_rose.py
from abc import abstractmethod, ABC
from pydantic import BaseModel


class Item(BaseModel):
    name : str
    sell_in : int
    quality : int

class GildedRose(ABC):
    def __init__(self, items: Item):
        self.name = items.name
        self.quality = items.quality
        self.sell_in = items.sell_in

    @abstractmethod
    def update_quality(self):
        pass

File: Java-Approvals/src/main/java/com/gildedrose/GildedRose.java
package com.gildedrose;

class GildedRose {

    Item[] items;

    public GildedRose(Item[] items) {
        this.items = items;
    }

    public void updateQuality() {
        for (Item item : items) {
            if (item.name.equals(Item.AGED_BRIE)) {
                item.increaseQualityByOne();
            } else if (item.name.equals(Item.BACKSTAGE_PASSES)) {
                item.increaseQualityBackstage();
            } else {
                item.decreaseQualityByOne();
            }

            item.decreaseSellInEachDay();

            if (item.sellIn < 0) {
                if (item.name.equals(Item.AGED_BRIE)) {
                    item.increaseQualityByOne();
                } else if (item.name.equals(Item.BACKSTAGE_PASSES)) {
                    item.quality = 0;
                } else {
                    item.decreaseQualityByOne();
                }
            }
        }
    }

}
